src/Algolia.Search.Test/EndToEnd/Index/MultiClusterManagementTest.cs

```diff
@@ -60,14 +60,29 @@ public async Task McmTest()
 
             RemoveUserId(userId);
 
-            ListUserIdsResponse listUserIdsTwo = await BaseTest.McmClient.ListUserIdsAsync();
-            var yesterday = DateTime.UtcNow.AddDays(-1).ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
-            IEnumerable<UserIdResponse> userIdsToRemove =
-                listUserIdsTwo.UserIds.Where(x => x.UserID.Contains($"csharp-{yesterday}"));
-
-            var delete =
-                userIdsToRemove.Select(x => BaseTest.McmClient.RemoveUserIdAsync(x.UserID)).ToArray();
-            await Task.WhenAll(delete);
+            await RemovePastUserIDs();
+        }
+
+        private async Task RemovePastUserIDs()
+        {
+            const int hitsPerPage = 100;
+            int page = 0;
+            var userIDsToRemove = new List<UserIdResponse>();
+
+            while (true)
+            {
+                var response = await BaseTest.McmClient.ListUserIdsAsync(0,100);
```

Comment:
Use the `hitsPerPage` and `page` variables here instead of the hardcoded `0` and `100` values.

Reply:
Arg! Thanks for catching it. Solved with last commit.